forge-game/src/main/java/forge/game/cost/CostPayment.java

```diff
@@ -229,8 +229,8 @@ public final boolean payComputerCosts(final CostDecisionMakerBase decisionMaker)
      * @return a {@link forge.game.mana.Mana} object.
      */
     public static Mana getMana(final Player player, final ManaCostShard shard, final SpellAbility saBeingPaidFor,
-            final byte colorsPaid, Map<String, Integer> xManaCostPaidByColor) {
-        final List<Pair<Mana, Integer>> weightedOptions = selectManaToPayFor(player.getManaPool(), shard,
+            final byte colorsPaid, Map<String, Integer> xManaCostPaidByColor) { // player.getManaPool() is not threadsafe if this is called somwewhere concurrently
+        final List<Pair<Mana, Integer>> weightedOptions = selectManaToPayFor(new ManaPool(player), shard,
```

Comment:
you're just breaking AI ability to pay from pool completely because you're now calling this with new empty one??

Reply:
how come? they should be tally or it will not cast anything. Is there anything that can be reproducible with this change?

Reply:
Please read the code...
if you call `selectManaToPayFor` with an always empty mana pool `weightedOptions` will always be empty
**this function can no longer return anything except null**

Reply:
I rolled back it a while ago